app/src/main/java/com/example/weather/network/WeatherApiService.kt

```diff
@@ -0,0 +1,40 @@
+package com.example.weather.network
+
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+import retrofit2.http.GET
+
+private const val BASE_URL = "https://api.openweathermap.org/"
+
+
+/**
+ * Build the Moshi object with Kotlin adapter factory that Retrofit will be using.
+ */
+private val moshi = Moshi.Builder()
+    .add(KotlinJsonAdapterFactory())
+    .build()
+
+/**
+ * The Retrofit object with the Moshi converter.
+ */
+private val retrofit = Retrofit.Builder()
+    .addConverterFactory(MoshiConverterFactory.create(moshi))
+    .baseUrl(BASE_URL)
+    .build()
+
+/**
+ * A public interface that exposes the [getCities] method
+ */
+interface WeatherApiService {
+    @GET("data/2.5/group?id=4772354,5368361,5128581,4887398,4699066,5809844,5856195,5419384,5391959,4684888,5746545,4990729,4335045&appid=bd4472d97ca0b479dc32513cf50a1bf3&units=imperial.json")
```

Comment:
We don't want to hardcode the entire route here in this @GET annotation. Instead, we want this `getCities` method to accept these route queries as parameters. Then, we can pass in the ids, the appId, and the units from the viewModel. This makes it so this method is much more versatile and can be used to fetch a wide range of id's/units/etc. If you need some tips on this feel free to reach out to me